nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/GetAzureEventHub.java

```diff
@@ -332,7 +385,13 @@ private EventHubConsumerClient createEventHubConsumerClient(final ProcessContext
             final AzureNamedKeyCredential azureNamedKeyCredential = new AzureNamedKeyCredential(policyName, policyKey);
             eventHubClientBuilder.credential(fullyQualifiedNamespace, eventHubName, azureNamedKeyCredential);
         }
-        return eventHubClientBuilder.buildConsumerClient();
+
+        // Set Azure Event Hub Client Identifier using Processor Identifier instead of default random UUID
+        final AmqpClientOptions clientOptions = new AmqpClientOptions();
+        clientOptions.setIdentifier(getIdentifier());
```

Comment:
The processor's identifier is the same on all cluster nodes so all instances would use the same id.
When the processor is configured to run in "All nodes" mode, it can be confusing to have multiple clients with the same identifier. Also, in "Primary node" mode it would be useful to see which node the client belongs to.
So I suggest adding some node-specific tag in the identifier.